sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/persons/PersonDirectorySteps.java

```diff
@@ -106,25 +211,83 @@ public PersonDirectorySteps(
                       + " "
                       + apiState.getLastCreatedPerson().getFirstName();
               break;
-              // etc
+            case "phone number":
+              searchText = apiState.getLastCreatedPerson().getPhone();
+              break;
+            case "email":
+              searchText = apiState.getLastCreatedPerson().getEmailAddress();
+              break;
           }
-          webDriverHelpers.waitUntilElementIsVisibleAndClickable(APPLY_FILTERS_BUTTON);
-          webDriverHelpers.clickOnWebElementBySelector(ALL_BUTTON);
-          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(150);
 
           webDriverHelpers.fillInWebElement(MULTIPLE_OPTIONS_SEARCH_INPUT, searchText);
+        });
+
+    Then(
+        "I change id,full name,email,phone number, by {string}",
```

Comment:
1. divide this logic into separated steps/methods to make the code reusable, and add also the location of the action